### ISSUE TITLE
User reports crash on creating new LUKS + LVM

### ISSUE THREAD

**Issue by geckolinux** [calamares/calamares#691]
Hello, a GeckoLinux user reported the following crash on both Calamares 3.0 and 3.1 upon trying to install to a new LUKS + LVM in a virtual machine:
https://github.com/geckolinux/geckolinux-project/files/788363/Calamares.log.txt

**Comment by highvoltage** [comment]
Do you have cryptsetup installed on your live media?

**Comment by geckolinux** [comment]
Hello @highvoltage , yep, `cryptsetup` is there.

What exactly is expected to work and not work regarding LUKS and LVM in Calamares? Full disk and/or single partition(s)? Partitions newly created with Calamares and/or pre-existing ones?

**Comment by Ragebone** [comment]
Hello,
I too can't install Manjaro 17.0 with encryption.
Fully automatic install, checkbox for encryption, and it failes with:

`The Installer failed to create file system on partition /dev/nvme0n1p2`
`Command: cryptsetup -s 512 --batch-mode ..force-password luksFormat /dev/nvme0n1p2`
`Command: cryptsetup open /dev/nvme0n1p2 luks-............................`
`Command: mkfs.ext4 -qf /dev/mapper/nvme0n1p2`

at about 20% of progress, pretty fast after starting.

System:
Intel i7 6700k
Asus Z170 ws
Samsung 950 pro 500GB
16GB RAM.

And the install USBstick.

Calamares is at version 3.1.0-3

Thanks!

**Comment by billwanjohi** [comment]
I am also experiencing this issue while attempting an install of Manjaro i3 17.0.1 via Calamares 3.1.0.

It seems like one obvious problem is that the cryptsetup command is specifying some unique volume name, but then the subsequent mkfs is trying to refer to the logical, decrypted volume with the name of the device. I get the same error message as Ragebone, and after the failure can manually format and mount the logical device that Calamares successfully created and opened. That's to no avail, because the Calamares partition manager ignores that volume, only offering to replace the original volume, bringing me full circle.

Unfortunately, my C++ is not strong enough (and the debug logging not verbose enough) for me to determine where this is happening, let alone submit a fix.

**Comment by billwanjohi** [comment]
On closer inspection, it looks like geckolinux's user has a different issue, because in their log it's the `luksFormat` command that's failing. I think I also figured out my and Ragebone's issue, and filed as a bug on the kpmcore project, which I believe is where the fault lies: https://bugs.kde.org/show_bug.cgi?id=379205

**Comment by dukebarman** [comment]
I have the same issue with installing Manjaro KDE 17.0.1

**Comment by philmmanjaro** [comment]
Please check if this is fixing it:

```
sudo pacman -U http://repo.manjaro.org.uk/pool/overlay/kpmcore-3.0.3-1.1-$(uname -m).pkg.tar.xz
sudo -E calamares -d
```

**Comment by dukebarman** [comment]
Installiation was completed. But after reboot doesn't accept right password and can't find partition (enter grub rescue). Without encrypt installiation works. Maybe this bug like a https://github.com/calamares/calamares/issues/561?

**Comment by philmmanjaro** [comment]
For debuging we always need the full log. You can start **calamares** always thru a terminal with `sudo -E calamares -d` and use **pastebin** or **gist** to share the full terminal output. We only tested **automatic** partitioning with **luks** encrytion so far. The **manual** way might still be broken.

**Comment by stikonas** [comment]
Most likely fixed in kpmcore 3.1.0

**Comment by adriaandegroot** [comment]
.. and with the better LUKS configuration in Calamares 3.1.4, so I'm going to close this as "probably fixed".
